pytorch_lightning/core/optimizer.py

```diff
@@ -337,15 +337,26 @@ def _get_default_scheduler_config() -> Dict[str, Any]:
         "reduce_on_plateau": False,  # most often not ReduceLROnPlateau scheduler
         "monitor": None,  # value to monitor for ReduceLROnPlateau
         "strict": True,  # enforce that the monitor exists for ReduceLROnPlateau
-        "opt_idx": None,  # necessary to store opt_idx when optimizer frequencies are specified
+        "opt_idx": None,  # opt_idx assigned internally if not assigned by user
     }
 
 
-def _validate_scheduler_optimizer(optimizers: List[Any], lr_schedulers: List[Any]) -> None:
-    if any(sch["scheduler"].optimizer not in optimizers for sch in lr_schedulers):
-        raise MisconfigurationException(
-            "Some schedulers are attached with an optimizer that wasn't returned from `configure_optimizers`."
-        )
+def _attach_scheduler_opt_idx(optimizers: List[Any], lr_schedulers: List[Any]) -> None:
```

Comment:
```suggestion
def _set_scheduler_opt_idx(optimizers: List[Any], lr_schedulers: List[Any]) -> None:
```
?

"attach" usually refers to setting a reference to an object or something like that.